sdcm/cluster.py

```diff
@@ -3771,35 +3774,22 @@ def critical_node_setup_events():
             add_severity_limit_rules([f'{e["rule_pattern"]}={e["default_severity"]}' for e in critical_events])
 
         start_time = time.perf_counter()
-        init_nodes = []
-        results = []
 
         if isinstance(cl_inst, BaseScyllaCluster):
             # Update installed scylla before node setup, scylla server will be start at the end of node_setup
             cl_inst.update_db_binary(node_list, start_service=False)
             cl_inst.update_db_packages(node_list, start_service=False)
 
         with critical_node_setup_events():
+            start_time = time.perf_counter()
+            cl_inst.run_func_parallel(func=node_setup, node_list=node_list)
```

Comment:
worth to verify results if there are no exceptions - so could fail fast before nodes startup

Reply:
`node_setup` function has `@raise_event_on_failure` decorator which catches all the exceptions, creates an SCT event based on it and returns `None`. So, in the `run_func_parallel` method we always will not have exceptions.

Reply:
no, `node_setup` does catch all the exceptions and put is as element in tuple to queue - and we ignore later verifying that queue if exceptions are there

Reply:
Look at the mentioned decorator code, it wraps the exceptions itself.
In other words, `try`-block in the `node_setup` won't see any exceptions due to the decorator.

Reply:
still I don't get how already caught and ignored exception can be caught again by decorator

Reply:
@soyacz you are right. My mistake. It is to be fixed.

Reply:
Done